### PR TITLE
Solutions

### PR DESCRIPTION
**Pull request by gaupt**

### CODE REVIEW

File: network.tf [hunk 1]
```diff
@@ -0,0 +1,35 @@
+resource "azurerm_resource_group" "ts4" {
+  name     = "${var.prefix}-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_virtual_network" "main" {
+  name                = "${var.prefix}-network"
+  address_space       = ["10.0.0.0/16"]
```
Comment (NazarSheremet) [inline network.tf:8]: It would be better to move this value to variables

File: network.tf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  location            = azurerm_resource_group.ts4.location
+  resource_group_name = azurerm_resource_group.ts4.name
+}
+
+resource "azurerm_subnet" "internal" {
+  name                 = "internal"
+  resource_group_name  = azurerm_resource_group.ts4.name
+  virtual_network_name = azurerm_virtual_network.main.name
+  address_prefixes     = ["10.0.2.0/24"]
```
Comment (NazarSheremet) [inline network.tf:17]: It would be better to move this value to variables

File: network.tf [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+locals {
+  nic_names = ["nic1", "nic2", "nic3"]
+}
```
Comment (NazarSheremet) [inline network.tf:22]: It would be better to create a separate locals.tf file and move this value into it.

File: network.tf [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  name                = "${var.prefix}-nic-${count.index + 1}"
+  location            = azurerm_resource_group.ts4.location
+  resource_group_name = azurerm_resource_group.ts4.name
+
+  ip_configuration {
+    name                          = "testconfiguration1"
+    subnet_id                     = azurerm_subnet.internal.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
```
Comment (NazarSheremet) [inline network.tf:35]: Add EoF
image

File: variables.tf [hunk 5]
```diff
@@ -0,0 +1,7 @@
+variable "prefix" {
+  default = "tfvmex"
+}
+
+variable "prod_is" {
+  default = false
+}
```
Comment (NazarSheremet) [inline variables.tf:7]: Need to add description and type for variables

File: vmss.tf [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    prevent_destroy = true
+  }
+}
+
+locals {
+  security_rules = [
+    { name = "rule1", priority = 100, direction = "Inbound", access = "Allow", protocol = "*", source_port_range = "*", destination_port_range = "22", source_address_prefix = "*", destination_address_prefix = "*" },
+    { name = "rule2", priority = 200, direction = "Inbound", access = "Allow", protocol = "*", source_port_range = "*", destination_port_range = "80", source_address_prefix = "*", destination_address_prefix = "*" }
+  ]
+}
```
Comment (NazarSheremet) [inline vmss.tf:42]: Move this locals to the newly created file `locals.tf` too

**Review by NazarSheremet — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 5 above)
Comment (gaupt) [reply variables.tf:7]: видає помилку terraform fmt -check -recursive -diff

File: vmss.tf (hunk 6 above)
Comment (gaupt) [reply vmss.tf]: fix

File: network.tf (hunk 4 above)
Comment (gaupt) [reply network.tf]: fix

File: network.tf (hunk 3 above)
Comment (gaupt) [reply network.tf]: fix

File: network.tf (hunk 1 above)
Comment (gaupt) [reply network.tf]: fix (add to variables)

File: network.tf (hunk 2 above)
Comment (gaupt) [reply network.tf]: fix (add to variavles)

File: variables.tf (hunk 5 above)
Comment (gaupt) [reply variables.tf]: make only variable "prefix" {
  default = "tfvmex"
}

**Review by NazarSheremet — APPROVED** [review]
Good job!

**Pull request closed without merging**